packages/central-server/src/apiV2/surveyResponses/GETSurveyResponses.js

```diff
@@ -20,6 +20,18 @@ import { getQueryOptionsForColumns } from '../GETHandler/helpers';
 export class GETSurveyResponses extends GETHandler {
   permissionsFilteredInternally = true;
 
+  customJoinConditions = {
+    country: {
+      through: 'entity',
+      foreignKey: 'entity.country_code',
+      foreignTable: 'country.code',
```

Comment:
I really like `through`! I think the other two fields could do with clearer names though. When we're going `through` another table, there are up to four columns we have to deal with:
- The column we launch off from the base table this endpoint represents 
- The column that the base table column connects to in the through table
- The column in the through table that we connect further on with
- The column in the final join table that we connect to

I wouldn't be able to distinguish which column either `foreignKey` or `foreignTable` relates to, or what would happen if I needed to specify rather than use the default for all four columns. Perhaps have a chat to @EMcQ-BES about alternative names that make this all clear - remember that code is read a lot more than it is written, so we take time now to optimise for people to understand it easily later on!

Reply:
@edmofro Chatting with @WDuncker about this. I figure we can't _quite_ call them something like e.g. `localTableKey` and `foreignTableKey` because when connecting with a `through` table the first key (`entity.country_code` in this case) technically _isn't_ local to the original table. That variable is playing a bit of double duty here.

Wim floated `nearTableKey` and `farTableKey` which I reckon describes the two pieces of behaviour pretty well. But I figure it might be worth getting a final opinion from you as a more external viewpoint

Reply:
That double duty is a bit what I'm worried about @EMcQ-BES. We don't currently have a requirement where you have to specify non-default columns on all four possible places (the local table from, the through table to, the through table from, and final table to), but what if you did? It feels a bit messy/unpredictable that when you define `farTableKey` you don't know whether it'll be used in the through or the final table.

Perhaps we need `through` to take an object, which has `localTableKey` and `foreignTableKey` as well as `table`?

Reply:
If you need all four columns to be custom you'd just add another `customJoin`. They chain together in the recursive function. The `farTableKey` is always against the table we're joining to with that definition.

```
Example GETDashboards.js

customJoins: {
  entity: {
    nearTableKey: "dashboard.root_entity_code",
    farTableKey: "entity.code",
  },
  country: {
    through: "entity",
    nearTableKey: "entity.country_code",
    farTableKey: "country.code",
  },
}
```

I feel like I might be missing something in your question. Perhaps we can chat about this tomorrow

Reply:
Right, this makes sense! In that case, I think the names are as good as you'd get. Thanks for bearing with me :-)